### PR TITLE
add template documentation

### PR DESCRIPTION
**Pull request by anludke**

### CODE REVIEW

File: documentation-templates/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# documentation-templates
+
+The templates available in [scaffolder-templates](https://github.com/parodos-dev/workflow-software-templates/tree/main/scaffolder-templates) are integrated in the Backstage Orchestrator plugin under `locations` in the following deployment manifest [values](https://github.com/anludke/orchestrator-helm-chart/blob/main/charts/orchestrator/values.yaml#L121).
```
Comment (masayag) [inline documentation-templates/README.md:3]: pls update the link to the values to point to its new location
https://github.com/parodos-dev/orchestrator-helm-chart/blob/main/charts/orchestrator/templates/rhdh-operator.yaml#L186

**Review by RichieW98 — APPROVED** [review]

File: documentation-templates/README.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+## basic-workflow
+This template allows the user to generate a basic workflow project from the user's inputs collected.
+Based on the workflow type (assessment, infrastructure) the user selects, the corresponding workflow template is generated.
+
+### basic-workflow as assessment
```
Comment (masayag) [inline documentation-templates/README.md:9]: pls mention the expected output schema format of the assessment workflow which differentiates it from the infrastructure workflow and enables it to be shown in the UI.

File: documentation-templates/README.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+The assessment workflow project generated from the basic-workflow template is a simple workflow that evaluates the user's input as string by comparing it to the word `default` to print out the default workflow options or an empty workflow options list.
+
+### basic-workflow as infrastructure
+The infrastructure workflow project generated from the basic-workflow template is a simple workflow that prints out the workflow's data input.
+
+
+## complex-assessment-workflow
+This template allows the user to generate a complex assessment workflow project from the user's inputs collected. It includes:
+1. **custom java code** to perform complex operations on the user's inputs. In this case, it evaluates the user's input as string by comparing it to the word `default` to print out the default workflow options or an empty workflow options list from the java code.
```
Comment (masayag) [inline documentation-templates/README.md:18]: should we add some requirements to run the generated workflows locally or is it explained somewhere else?

**Review by masayag — CHANGES_REQUESTED** [review]

File: documentation-templates/README.md (hunk 2 above)
Comment (anludke) [reply documentation-templates/README.md:9]: Done.

File: documentation-templates/README.md (hunk 1 above)
Comment (anludke) [reply documentation-templates/README.md]: Fixed.

File: documentation-templates/README.md (hunk 3 above)
Comment (anludke) [reply documentation-templates/README.md:36]: The generated workflow project comes with a README file with those instructions. But I added a 4. line with that information.

**Review by masayag — APPROVED** [review]

**Pull request merged**